index.js

```diff
@@ -179,7 +180,7 @@ module.exports = {
 
     var ui = this.ui;
 
-    return jshintTrees(tree, {
+    return jshintTrees(new Funnel(tree, { include: [/\.js$/] }), {
```

Comment:
TL;DR, we can easily decompose globs, and use the various glob segments to prune dir trees that could not possible contain matched files. With RegExp, that optimization is much more complicated (at-least in the general case)